src/meili/map_matcher.cc

```diff
@@ -277,31 +277,43 @@ MapMatcher::OfflineMatch(
     const std::vector<Measurement>& measurements, uint32_t k) {
   mapmatching_.Clear();
 
-  const auto begin = measurements.begin(),
-               end = measurements.end();
-
-  if (begin == end) {
+  if (measurements.size() < 2)
     return {};
-  }
 
   const auto interpolation_distance = config_.get<float>("interpolation_distance"),
           sq_interpolation_distance = interpolation_distance * interpolation_distance;
   std::unordered_map<StateId::Time, std::vector<Measurement>> interpolated_measurements;
 
-  // Always match the first measurement
-  auto time = AppendMeasurement(*begin);
-  auto latest_match_measurement = begin;
-  for (auto measurement = std::next(begin); measurement != end; measurement++) {
-    const auto sq_distance = GreatCircleDistanceSquared(*latest_match_measurement, *measurement);
-    // Always match the last measurement
-    if (sq_interpolation_distance < sq_distance || std::next(measurement) == end) {
-      time = AppendMeasurement(*measurement);
-      latest_match_measurement = measurement;
-    } else {
-      interpolated_measurements[time].push_back(*measurement);
-    }
+  // Since we always match the first and last measurement we need to find the first
+  // measurement from the end which will not be interpolated, the rest of them will
+  // be interpolated, so we keep track of the start of the run of these interpolated ones
+  size_t start_of_run = measurements.size() - 1;
+  for(; start_of_run > 0; --start_of_run) {
```

Comment:
youre right, and the tests showed it as well. it needs to keep track of the last one that its keeping not `i-1` i'm working on fixing it